#! /bin/sh /usr/share/dpatch/dpatch-run
## fix-479467.dpatch from Stephan Beyer <s-beyer@gmx.net>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix segmentation fault when pressing g in full screen mode

@DPATCH@
diff -ruN xpdf-3.02.old/xpdf/XPDFViewer.cc xpdf-3.02.fix/xpdf/XPDFViewer.cc
--- xpdf-3.02.old/xpdf/XPDFViewer.cc	2007-02-27 23:05:52.000000000 +0100
+++ xpdf-3.02.fix/xpdf/XPDFViewer.cc	2008-05-04 21:29:34.000000000 +0200
@@ -834,6 +834,9 @@
 
 void XPDFViewer::cmdFocusToPageNum(GString *args[], int nArgs,
 				   XEvent *event) {
+  if (core->getFullScreen()) {
+    return;
+  }
   XmTextFieldSetSelection(pageNumText, 0,
 			  strlen(XmTextFieldGetString(pageNumText)),
 			  XtLastTimestampProcessed(display));
